src/link/object.c

```diff
@@ -331,9 +338,8 @@ obj_ReadOpenFile(FILE * pObjfile, char *tzObjectfile)
 	tzHeader[4] = 0;
 	if (strncmp(tzHeader, "RGB", 3) == 0) {
 		switch (tzHeader[3]) {
-		case '3':
-		case '4': // V4 supports OAM sections, but is otherwise identical
-			obj_ReadRGB(pObjfile);
+		case '5':
```

Comment:
I see your point. The problem is that not checking this could cause the linker to crash without any useful error message to tell you what's wrong (for example, if you update RGBDS and don't clean before trying to compile). Just by changing the number (maybe even using characters A-Z and a-z) you get something that can be understood.